### PR TITLE
internal: Improve version handling for numbagg

### PR DESCRIPTION
**Pull request by max-sixty**
Uses the approach in #8316, a bit nicer. Only internal.

### CODE REVIEW

File: xarray/core/rolling_exp.py [hunk 1]
```diff
@@ -99,17 +100,17 @@ def __init__(
         window_type: str = "span",
         min_weight: float = 0.0,
     ):
-        if has_numbagg is False:
+        if _NUMBAGG_VERSION is None:
             raise ImportError(
                 "numbagg >= 0.2.1 is required for rolling_exp but currently numbagg is not installed"
             )
-        elif has_numbagg < "0.2.1":
```
Comment (dcherian) [inline xarray/core/rolling_exp.py:106]: Shall we just require a minimum of `0.3.1`?

Comment (max-sixty) [reply xarray/core/rolling_exp.py:106]: I had this originally, but we have an automated check for versions available 12 months ago. And the convention makes sense to me, so I thought this was OK.

I'm also adding lots of other methods; e.g. https://github.com/pydata/xarray/pull/8307, so we're going to have some months of version proliferation regardless...

File: xarray/core/rolling_exp.py [hunk 2]
```diff
@@ -100,17 +101,17 @@ def __init__(
...
+        elif _NUMBAGG_VERSION < Version("0.2.1"):
             raise ImportError(
-                f"numbagg >= 0.2.1 is required for `rolling_exp` but currently version {has_numbagg} is installed"
+                f"numbagg >= 0.2.1 is required for rolling_exp but currently version {_NUMBAGG_VERSION} is installed"
             )
-        elif has_numbagg < Version("0.3.1") and min_weight > 0:
+        elif _NUMBAGG_VERSION < Version("0.3.1") and min_weight > 0:
             raise ImportError(
-                f"numbagg >= 0.3.1 is required for `min_weight > 0` within `.rolling_exp` but currently version {has_numbagg} is installed"
+                f"numbagg >= 0.3.1 is required for `min_weight > 0` but currently version {_NUMBAGG_VERSION} is installed"
```
Comment (max-sixty) [inline xarray/core/rolling_exp.py:113]:
```suggestion
                f"numbagg >= 0.3.1 is required for `min_weight > 0` within `.rolling_exp` but currently version {_NUMBAGG_VERSION} is installed"
```

**Comment by max-sixty** [comment]
I actually made this change on another PR (I was wondering why it seemed like I was writing code I'd already written!), so this ends up being a tiny variable naming change. I'll still merge.

**Pull request merged**